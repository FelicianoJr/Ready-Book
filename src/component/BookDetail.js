import React from 'react'
import SelectBookOptions from './SelectBookOptions'

const BookDetail = ({ book, update }) => {

    return (
        <li>
            <div className="book">
                <div className="book-top">
                    <div className="book-cover"
                        style={{
                            width: 128, height: 193,
                            backgroundImage: `url(${book.imageLinks.thumbnail})`
                        }}>
                    </div>
                    <div className="book-shelf-changer">
                        <SelectBookOptions book={book} update={update} />
                    </div>
                </div>
                <div className="book-title">{book.title}</div>
                <div className="book-authors">{book.authors}</div>
            </div>
        </li>
    )
};

export default BookDetail;