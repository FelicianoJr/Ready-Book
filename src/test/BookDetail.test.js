import BookDetail from '../component/BookDetail'
import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

describe("<BookDetail/>", () => {

    let update;
    let props;

    beforeEach(() => {
        update = jest.fn();
        props = {
            book: {
                id: 0, imageLinks: { thumbnail: 'url/test/enzime' },
                title: 'React', shelf: 'read'
            },
            update: update,
        };
    });

    it("should contain the imageLinks passed as props", () => {
        const wrapper = mount(<BookDetail {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot();
        expect(wrapper.props().book.imageLinks.thumbnail).toEqual('url/test/enzime');
    });

    it("should contain the Title passed as props", () => {
        const wrapper = mount(<BookDetail {...props} />)
        expect(wrapper.props().book.title).toEqual('React');
    });

    it('should render six <div>', () => {
        const wrapper = shallow(<BookDetail {...props} />)
        expect(wrapper.find('div').length).toEqual(6);
    });

    it('should render one <il>', () => {
        const wrapper = shallow(<BookDetail {...props} />)
        expect(wrapper.find('li').length).toEqual(1);
    });

})