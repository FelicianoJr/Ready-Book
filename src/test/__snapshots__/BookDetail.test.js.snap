exports[`<BookDetail/> should contain the imageLinks passed as props 1`] = `
<BookDetail
  book={
    Object {
      "id": 0,
      "imageLinks": Object {
        "thumbnail": "url/test/enzime",
      },
      "shelf": "read",
      "title": "React",
    }
  }
  update={[Function]}>
  <li>
    <div
      className="book">
      <div
        className="book-top">
        <div
          className="book-cover"
          style={
            Object {
              "backgroundImage": "url(url/test/enzime)",
              "height": 193,
              "width": 128,
            }
          } />
        <div
          className="book-shelf-changer">
          <SelectBookOptions
            book={
              Object {
                "id": 0,
                "imageLinks": Object {
                  "thumbnail": "url/test/enzime",
                },
                "shelf": "read",
                "title": "React",
              }
            }
            update={[Function]}>
            <select
              onChange={[Function]}
              value="read">
              <option
                disabled={true}
                value="">
                Move to...
              </option>
              <option
                value="currentlyReading">
                Currently Reading
              </option>
              <option
                value="wantToRead">
                Want to Read
              </option>
              <option
                value="read">
                Read
              </option>
              <option
                value="">
                None
              </option>
            </select>
          </SelectBookOptions>
        </div>
      </div>
      <div
        className="book-title">
        React
      </div>
      <div
        className="book-authors" />
    </div>
  </li>
</BookDetail>
`;
